packages/main/src/plugin/menu-registry.ts

```diff
@@ -0,0 +1,48 @@
+/**********************************************************************
+ * Copyright (C) 2022 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+export interface Menu {
+  command: string;
+  title: string;
+}
+
+export class MenuRegistry {
+  private menus = new Map<string, Menu[]>();
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  registerMenus(menus: any): void {
```

Comment:
I don't see why the menu should be of any type here

Reply:
Before the way the data is structured makes it impossible to type as the property names are open and are equal to the context names

Reply:
we can enforce the type to be Menu[]

Reply:
menus is with the type `[key: string]: Menu[];` no ?

Reply:
Needs to improve my Typescript tips knowledge base :). Done

Reply:
thanks 👍